aggregatedpool/workers.go

```diff
@@ -86,6 +86,10 @@ func TemporalWorkers(wDef *Workflow, actDef *Activity, wi []*internal.WorkerInfo
 		}
 
 		for j := 0; j < len(wi[i].Activities); j++ {
```

Comment:
You may check for the `disableActivityWorkers` before the loop and use goto <some-var> to jump directly to the `workers = append(workers, wrk)` to skip useless iteration.